File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/components/ParticleEffectComponentJson.java
package io.github.movementspeed.nhglib.data.models.serialization.components;

import com.badlogic.gdx.utils.JsonValue;
import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson;
import io.github.movementspeed.nhglib.runtime.ecs.components.graphics.ParticleEffectComponent;

/**
 * Created by Fausto Napoli on 19/12/2016.
 */
public class ParticleEffectComponentJson extends ComponentJson {
    @Override
    public void parse(JsonValue jsonValue) {
        ParticleEffectComponent particleEffectComponent =
                nhg.entities.createComponent(entity, ParticleEffectComponent.class);

        particleEffectComponent.asset = jsonValue.getString("asset", "");
        output = particleEffectComponent;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/components/ParticleEffectComponentJson.kt
package io.github.movementspeed.nhglib.data.models.serialization.components

import com.badlogic.gdx.utils.JsonValue
import io.github.movementspeed.nhglib.core.ecs.components.graphics.ParticleEffectComponent
import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson

/**
 * Created by Fausto Napoli on 19/12/2016.
 */
class ParticleEffectComponentJson : ComponentJson() {
    override fun parse(jsonValue: JsonValue) {
        val particleEffectComponent = nhg!!.entities.createComponent(entity, ParticleEffectComponent::class.java)

        particleEffectComponent.asset = jsonValue.getString("asset", "")
        output = particleEffectComponent
    }
}
